dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java

```diff
@@ -57,67 +57,120 @@
-            .requiredPortlet("roles")
-            .init();
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, null)
+                .rejectWhenNoUser(true)
+                .requiredPortlet("roles")
+                .init();
 
-    Response response = null;
+        Response response = null;
 
-    try {
+        try {
+            final String portletId = portletApi.portletIdPrefixCleaner(formData.portletId);
+            if (UtilMethods.isSet(portletApi.findPortlet(portletId))) {
+                throw new DoesNotExistException("Portlet with Id: " + formData.portletId + " already exist");
+            }
 
-      final Portlet contentPortlet = portletApi.findPortlet("content");
+            final Portlet contentPortlet = portletApi.findPortlet("content");
 
-      final Map<String, String> initValues = new HashMap<>(contentPortlet.getInitParams());
-      initValues.put("name", formData.portletName);
-      initValues.put("baseTypes", formData.baseTypes);
-      initValues.put("contentTypes", formData.contentTypes);
-      initValues.put(DATA_VIEW_MODE_KEY, formData.dataViewMode);
+            final Map<String, String> initValues = new HashMap<>(contentPortlet.getInitParams());
+            initValues.put("name", formData.portletName);
+            initValues.put("baseTypes", formData.baseTypes);
+            initValues.put("contentTypes", formData.contentTypes);
+            initValues.put(DATA_VIEW_MODE_KEY, formData.dataViewMode);
 
-      final Portlet newPortlet = APILocator.getPortletAPI()
-          .savePortlet(new DotPortlet(formData.portletId, contentPortlet.getPortletClass(), initValues), initData.getUser());
+            final Portlet newPortlet = APILocator.getPortletAPI()
+                    .createOrUpdatePortlet(new DotPortlet(portletId, contentPortlet.getPortletClass(), initValues), initData.getUser());
 
-      return Response.ok(new ResponseEntityView(map("portlet", newPortlet.getPortletId()))).build();
+            return Response.ok(new ResponseEntityView(map("portlet", newPortlet.getPortletId()))).build();
 
-    } catch (Exception e) {
-      response = ResponseUtil.mapExceptionResponse(e);
+        } catch (Exception e) {
+            response = ResponseUtil.mapExceptionResponse(e);
+        }
+
+        return response;
     }
 
-    return response;
-  }
+    /**
+     * Saves a new working version of an existing Portlet.
+     * The formData must contain the identifier of the Portlet.
+     * @param request
+     * @param formData
+     * @return
+     */
+    @PUT
+    @Path("/custom")
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response save(@Context final HttpServletRequest request, final CustomPortletForm formData) {
```

Comment:
rename it to updatePortlet instead of save, to easier understanding.

Reply:
Done https://github.com/dotCMS/core/pull/27986/commits/8f9eef068702ebb83612e9044280e1ea9f9ddd10